### PR TITLE
Refactor mintting logic

### PR DESCRIPTION
**Pull request by BrandonWeng**
## Describe your changes and provide context
Mint denom should just be usei as that's the base currency that the staking rewards will be paid out as, it shouldn't be any other denoms 

But we'll keep it in genesis to avoid confusion around the denom of the fields 

See inline comments for more detailed description 

## Testing performed to validate your change
unit tests 

image

### CODE REVIEW

File: x/mint/types/params.go [hunk 1]
```diff
@@ -73,15 +73,19 @@ func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
...
 	if !ok {
 		return fmt.Errorf("invalid parameter type: %T", i)
 	}
-
-	if strings.TrimSpace(v) == "" {
+	if denomTrimed == "" {
 		return errors.New("mint denom cannot be blank")
 	}
-	if err := sdk.ValidateDenom(v); err != nil {
+	if denomTrimed != "usei" {
```
Comment (philipsu522) [inline x/mint/types/params.go]: nit - is there a variable we can use that specifies base denom? maybe
```
x/dex/types/keys.go:155:        DefaultPriceDenom = "usei"
```

Comment (BrandonWeng) [reply x/mint/types/params.go]: gotcha, using the default bond denom

File: proto/mint/v1beta1/mint.proto [hunk 2]
```diff
@@ -5,14 +5,22 @@ option go_package = "github.com/sei-protocol/sei-chain/x/mint/types";
 
 import "gogoproto/gogo.proto";
 
-// Minter represents the minting state.
+// Minter represents the most recent
 message Minter {
-  // current epoch provisions
-  string epoch_provisions = 1 [
```
Comment (BrandonWeng) [inline proto/mint/v1beta1/mint.proto:11]: not used at all, going to change minter to track the state of a mint instead

File: proto/mint/v1beta1/mint.proto [hunk 3]
```diff
@@ -26,14 +34,8 @@ message Params {
 
   // type of coin to mint
   string mint_denom = 1;
-  // epoch provisions from the first epoch
-  string genesis_epoch_provisions = 2 [
```
Comment (BrandonWeng) [inline proto/mint/v1beta1/mint.proto:30]: not used at all

File: x/mint/client/cli/query.go [hunk 4]
```diff
@@ -62,8 +61,8 @@ func GetCmdQueryParams() *cobra.Command {
 // epoch provisions value.
 func GetCmdQueryEpochProvisions() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "epoch-provisions",
-		Short: "Query the current minting epoch provisions value",
+		Use:   "minter",
```
Comment (BrandonWeng) [inline x/mint/client/cli/query.go:64]: query the last minter saved, i.e the last mint event

**Review by philipsu522 — APPROVED** [review]

**Pull request merged**